src/main/java/com/alyrow/gdx/particle/physics/Fan.java

```diff
@@ -1,26 +1,49 @@
 package com.alyrow.gdx.particle.physics;
 
 import com.badlogic.gdx.math.Vector2;
+import com.badlogic.gdx.utils.Json;
+import com.badlogic.gdx.utils.JsonValue;
 
-public class Fan extends PhysicForce {
+public class Fan extends PhysicForce implements Json.Serializable {
 
-    // Black hole specific
     private float speed;
     private Vector2 center;
 
+    public Fan() {
+    }
+
     public Fan(float x, float y, float speed) {
         center = new Vector2(x, y);
-        this.speed = speed/100;
+        this.speed = speed / 100;
     }
 
-    private Vector2 cache = Vector2.Zero;
+    private Vector2 cache = new Vector2();
 
     @Override
     public Vector2 getForce(PhysicParticle particle) {
-        cache.x = particle.y - center.y;
-        cache.y = center.x - particle.x;
+        return cache.set(particle.y - center.y, center.x - particle.x).scl(speed);
+    }
+
+    @Override
+    public void write(Json json) {
+
+        json.writeValue("center x", center.x);
+        json.writeValue("center y", center.y);
+
+        json.writeValue("speed", speed);
 
-        return cache.scl(speed);
     }
 
+    @Override
+    public void read(Json json, JsonValue jsonData) {
+
+        center = new Vector2(jsonData.getFloat("center x"), jsonData.getFloat("center y"));
+        speed = jsonData.getFloat("speed");
+
+    }
```

Comment:
Need to review in details the serialization

Reply:
Such a great idea, when I will have finished optimisation, I will do that
https://github.com/alyrow/box2d-light-particle/commit/3529ee518b4d3a751cfdc9f7337f19b207e05fd2